ISM43362/ISM43362.cpp

```diff
@@ -435,20 +438,54 @@ int ISM43362::scan(WiFiAccessPoint *res, unsigned limit)
     char *ptr;
     char tmp[256];
     bool found = false;
-
-    if (!(_parser.send("F0"))) {
-        debug_if(_ism_debug, "\tISM43362: scan error\r\n");
-        return 0;
+    bool AP_Scan_1by1 = false;
+
+    /* Recent FW version provide new AT command: F0=2 which allows to solve the issue too mcuh AP
+      With too much AP (depending on SSID name length), there is no answer from firmware to F0 command.
+      F0=2 command allows to scan AP one by one. Command MR is then used to get next AP. */
+    if (_FwVersionId >= ParseNumber((char *)SINGLE_AP_SCAN_FW_VERSION_NUMBER, NULL)) {
+        AP_Scan_1by1 = true;
+    }
+    if (AP_Scan_1by1 == true) {
+        if (!(_parser.send("F0=2"))) {
+            debug_if(_ism_debug, "\tISM43362: scan error\r\n");
+            return 0;
+        }
+    } else {
+        if (!(_parser.send("F0"))) {
+            debug_if(_ism_debug, "\tISM43362: scan error\r\n");
+            return 0;
+        }
     }
 
+
     /* Parse the received buffer and fill AP buffer */
     /* Use %[^\n] instead of %s to allow having spaces in the string */
-    while (_parser.recv("#%[^\n]\n", tmp)) {
+    while (_parser.recv("%[^\n^\r]\r\n", tmp)) {
         found = false;
+
+
         if (res != 0 && limit != 0 && cnt >= limit) {
             /* reached end */
+            if (AP_Scan_1by1 == true) {
+                /* continue to receive AP 1 by 1 from firmware until we get OK */
+                while ((strncmp("OK\r", (char *)tmp, 2) != 0)) {
+                    _parser.flush();
+                    _parser.send("MR");
+                    _parser.recv("%[^\n^\r]\r\n", tmp);
```

Comment:
Maybe check for errors or timeouts in the send and recv to make sure we never end up stuck into this loop in case the firmware reports an error ?

Reply:
Done